### ISSUE TITLE
Make loss functions compatible with the sklearn api in xgboost/lightghm

### ISSUE THREAD

**Issue by sbjelogr** [orchardbirds/bokbokbok#76]
Currently bokbokbok implements the loss functions to be used in the python api of the main gradient boosting packages.

However, it would be very convenient to add also a version compatible with the sklearn API those packages.

The convention is as follows:

```python
my_amazing_loss_function(y_true, y_pred):

    ... do my magic

   return grad, hess
```

Do you agree @orchardbirds ?
